tests/tests/tier0/bluechictl-metrics/python/bluechictl_metrics.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import re
+import time
+import unittest
+
+from bluechi_machine_lib.bluechictl import BackgroundRunner, BluechiCtl, SimpleEventEvaluator
+
+node_foo_name = "node-foo"
+service_simple = "simple.service"
+
+
+class TestBluechictlMetrics(unittest.TestCase):
+
+    def setUp(self) -> None:
+        signal_agent_metrics_start_pat = re.compile(r"^.*Agent systemd StartUnit job.*")
+        signal_agent_metrics_stop_pat = re.compile(r"^.*Agent systemd StopUnit job.*")
+        signal_controller_metrics_pat = re.compile(r"^.*BlueChi job gross measured time.*")
+        expected_patterns = {signal_agent_metrics_start_pat,
+                             signal_agent_metrics_stop_pat,
+                             signal_controller_metrics_pat}
+        args = ["metrics", "listen"]
+        self.bgrunner = BackgroundRunner(args, SimpleEventEvaluator(expected_patterns))
+        self.bluechictl = BluechiCtl()
+
+    def test_bluechictl_metrics(self):
+        self.bluechictl.metrics_enable()
+        self.bgrunner.start()
+        time.sleep(0.5)
+        self.bluechictl.unit_start(node_foo_name, service_simple)
+        time.sleep(0.5)
+        self.bluechictl.unit_stop(node_foo_name, service_simple)
+        time.sleep(0.5)
+        self.bgrunner.stop()
```

Comment:
Hmm, wouldn't it be better to not set `self.finished = True` in `BackgroundRunner.stop()`? It seems to me that you need to set timeout before `bgrunner.stop()` only because you stop event processing loop. But shouldn't the event processing loop be stopped only with either "all required event found" or by `timeout_guard()`?

Reply:
I think the `timeout` should be saved to a failure of some sort, it's an overall timeout. Other than that, I want to be able to stop following output and see if I have or have not got the output I have expected or expected not to show

Reply:
You have no idea when events will be received, that's which we have the timeout_guard, which will be raised before the global test timeout, so interrrupting processing with timeout_guard will be able to collect all the output (because global test timeout will not preserve output from a process running on a controller).

Reply:
Understood. Done